node/consensus/src/lib.rs

```diff
@@ -431,6 +445,50 @@ impl<N: Network> Consensus<N> {
         Ok(())
     }
 
+    #[cfg(feature = "metrics")]
+    fn add_transmission_latency_metric(&self, next_block: &Block<N>) {
+        const AGE_THRESHOLD_SECONDS: i32 = 30 * 60; // 30 minutes set as stale transmission threshold
+
+        let mut keys_to_remove = Vec::new();
+        let mut transmission_queue_timestamps = self.transmissions_queue_timestamps.lock();
+
+        let solution_ids: std::collections::HashSet<_> = next_block.solutions().solution_ids().collect();
+        let transaction_ids: std::collections::HashSet<_> = next_block.transaction_ids().collect();
+
+        for (key, timestamp) in transmission_queue_timestamps.iter() {
+            let elapsed_time = std::time::Duration::from_secs((snarkos_node_bft::helpers::now() - *timestamp) as u64);
+
+            if elapsed_time.as_secs() > AGE_THRESHOLD_SECONDS as u64 {
+                // This entry is stale-- remove it from transmission queue and record it as a stale transmission
```

Comment:
```suggestion
                // This entry is stale - remove it from transmission queue and record it as a stale transmission.
```

Reply:
The M/double dash is correct here, but I can add a period :D